samples/instrumentation-quickstart/README.md

```diff
@@ -0,0 +1,67 @@
+# OpenTelemetry Fastify instrumentation example
+
+> [!NOTE]  
+> This sample is not an npm workspace within this repo. It does not link in any dependencies
+> from this repo and all commands should be run directly from this directory.
+
+This sample is a [Fastify](https://fastify.dev/) Node.js application instrumented with
+[OpenTelemetry JS auto instrumentation](https://opentelemetry.io/docs/languages/js/automatic/).
+This is a Node.js version of [this golang
+sample](https://github.com/GoogleCloudPlatform/golang-samples/tree/main/opentelemetry/instrumentation).
+It uses docker compose to orchestrate running the application and sending it some requests.
+
+The code is a basic Fastify application with two endpoints
+- `/multi` makes a few requests to `/single` on localhost
+- `/single` sleeps for a short time to simulate work
+
+Docker compose also runs the OpenTelemetry collector, set up to receive telemetry from the
+application and parse its logs from a shared volume. Finally, a loadgen container sends
+requests to the app.
+
+## Permissions
+
+This sample writes to Cloud Logging, Cloud Monitoring, and Cloud Trace. Grant yourself the
+following roles to run the example:
+- `roles/logging.logWriter` – see https://cloud.google.com/logging/docs/access-control#permissions_and_roles
+- `roles/monitoring.metricWriter` – see https://cloud.google.com/monitoring/access-control#predefined_roles
+- `roles/cloudtrace.agent` – see https://cloud.google.com/trace/docs/iam#trace-roles
+
+## Running the example
+
+### Cloud Shell or GCE
+
+```sh
+git clone https://github.com/GoogleCloudPlatform/opentelemetry-operations-js.git
+cd opentelemetry-operations-js/samples/instrumentation-quickstart
+docker compose up --abort-on-container-exit
+```
+
+### Locally with Application Default Credentials
+
+First Create local credentials by running the following command and following the
+oauth2 flow (read more about the command [here][auth_command]):
+
+	gcloud auth application-default login
+
+Set the `GOOGLE_APPLICATION_CREDENTIALS` environment variable with `export GOOGLE_APPLICATION_CREDENTIALS="$HOME/.config/gcloud/application_default_credentials.json"`
```

Comment:
Non-blocking: Isn't this already the default for ADC?  Do users actually need to set this?

Reply:
`docker-compose.creds.yaml` uses the environment variable as the source for the volume mount of credentials into the collector container. The compose file could default the environment variable to this, but the intention is to make it clear to the user that the sample will mount their creds. The path would also be different on Windows.

Do you think it would be better to default it?

Reply:
Got it.  So if we defaulted, the sample wouldn't work on windows?  If that is the case, the current implementation is good.

Reply:
Windows users would need to override the default. I haven't actually tried it on Windows though.

In case it wasn't clear, if they run on GCE or Cloud Shell they don't need to set any environment variables, just `docker compose up`